test/unit/lookup_key_test.rb

```diff
@@ -137,7 +137,9 @@ def test_value_should_not_be_changed
   end
 
   test "this is a smart class parameter?" do
-    assert lookup_keys(:complex).puppet?
+    assert_not_deprecated do
```

Comment:
I have added the deprecation test into this test because if I add it as another test, the ```LookupKey.all``` is already cached and will not raise the deprecation error.